tests/unit/Codeception/ConfigurationTest.php

```diff
@@ -67,4 +67,83 @@ public function testDefaultCustomCommandConfig()
+                    ['/My/proj/path/some/file/in/my/proj.txt',           'my/proj/path/',                '/',  '/My/proj/path/some/file/in/my/proj.txt'],
+                // Relative $path, Absolute projectDir()
+                    ['my/proj/path/some/file/in/my/proj.txt',            '/my/proj/path/',               '/',  'my/proj/path/some/file/in/my/proj.txt'],
+                // $path & projectDir() both relative
+                    ['my/proj/path/some/file/in/my/proj.txt',            'my/proj/path/foo/bar',         '/',  '../../some/file/in/my/proj.txt'],
+                // $path & projectDir() both absolute: not a subdir
+                    ['/my/proj/path/some/file/in/my/proj.txt',           '/my/proj/path/foobar/',        '/',  '../some/file/in/my/proj.txt'],
+                // ensure trailing DIRECTORY_SEPERATOR maintained
+                    ['/my/proj/path/some/dir/in/my/proj/',               '/my/proj/path/foobar',         '/',  '../some/dir/in/my/proj/'],
+            // Windows style paths:
+                // projectDir() with & without trailing directory seperator: actual subdir
+                    ['C:\\my\\proj\\path\\some\\file\\in\\my\\proj.txt', 'C:\\my\\proj\\path\\',         '\\', 'some\\file\\in\\my\\proj.txt'],
+                    ['C:\\my\\proj\\path\\some\\file\\in\\my\\proj.txt', 'C:\\my\\proj\\path',           '\\', 'some\\file\\in\\my\\proj.txt'],
+                    ['C:\\my\\proj\\pathsome\\file\\in\\my\\proj.txt',   'C:\\my\\proj\\path',           '\\', '..\\pathsome\\file\\in\\my\\proj.txt'],
+                // No device letter... absoluteness mismatch
+                    ['\\my\\proj\\path\\some\\file\\in\\my\\proj.txt',   'my\\proj\\path\\',             '\\',  '\\my\\proj\\path\\some\\file\\in\\my\\proj.txt'],
+                    ['my\\proj\\path\\some\\file\\in\\my\\proj.txt',     '\\my\\proj\\path\\',           '\\',  'my\\proj\\path\\some\\file\\in\\my\\proj.txt'],
+                // No device letter... absoluteness match
+                    ['my\\proj\\path\\some\\file\\in\\my\\proj.txt',     'my\\proj\\path\\foo\\bar',     '\\',  '..\\..\\some\\file\\in\\my\\proj.txt'],
+                    ['\\my\\proj\\path\\some\\file\\in\\my\\proj.txt',   '\\my\\proj\\path\\foobar\\',   '\\',  '..\\some\\file\\in\\my\\proj.txt'],
+                    ['\\my\\proj\\path\\some\\dir\\in\\my\\proj\\',      '\\my\\proj\\path\\foobar\\',   '\\',  '..\\some\\dir\\in\\my\\proj\\'],
+                // Device letter (both)... path absoluteness mismatch
+                    ['C:\\my\\proj\\path\\some\\file\\in\\my\\proj.txt', 'C:my\\proj\\path\\',           '\\',  '\\my\\proj\\path\\some\\file\\in\\my\\proj.txt'],
+                    ['d:my\\proj\\path\\some\\file\\in\\my\\proj.txt',   'd:\\my\\proj\\path\\',         '\\',  'my\\proj\\path\\some\\file\\in\\my\\proj.txt'],
```

Comment:
To my knowledge it is... It certainly was in the MSDOS days... Each drive letter keeps its own CWD